apis/python/src/tiledbsoma/_util.py

```diff
@@ -138,6 +138,17 @@ def validate_slice(slc: Slice[Any]) -> None:
     if slc.start is None or slc.stop is None:
         # All half-specified slices are valid.
         return
+
```

Comment:
given the behavior of `to_unix_ts`, couldn't you combine these two checks into one check (merging lines 142-155) - something like:

```
if to_unix_ts(slc.stop) < to_unix_ts(slc.start):
    raise ValueError(...)
```

Reply:
No because the values can be float

Reply:
If I read the code correctly, when they are float (or any type other than np.datetime64 or pa.TimestampScalar), `to_unix_ts` returns the value untouched (ie., the function is a noop).  So it ends up being exactly the same as the other if statement.

I certainly could be missing something - apologies if so.  But I don't think floats (or ints) are a problem here.

Reply:
`to_unix_ts` takes as arguments valid user timestamp inputs `np.datetime64`,`pa.TimestampScalar`, and `int`. The return type is `int` since it should always be a UNIX timestamp. I'd rather keep it like that than change the signature to be `Any` argument type and `Any` return type.